src/lib/coda_transition/external_transition.ml

```diff
@@ -119,12 +130,26 @@ Stable.Latest.
 
 include Comparable.Make (Stable.Latest)
 
+let (current_fork_id : Fork_id.t option ref) = ref None
```

Comment:
Minor nit, but this reference seems to make a little more sense to be stored in the `Fork_id` module itself rather than `External_transition`. Even though `External_transition` uses this value, it will not be the only place in the system where this value is referenced.

Reply:
As a second note, it may make more sense to make this a `Fork_id.t ref lazy_t` and putting the logic for loading the `Fork_id` from the filesystem into here. If it's too much trouble, it's not a big deal (annoyance here is giving this access to where the config directory is), but it would be nice from a code perspective since it would make it impossible to have `get_current_fork_id` raise an exception if you call it before calling `set_current_fork_id`.

Reply:
moved ref to `Fork_id`

Reply:
don't see easy way to get the conf dir over to `Fork_id`, except maybe by setting another ref, which is still a can of worms. So I'll stick with current strategy, which loads the fork id along with other config items at daemon startup.

Reply:
> don't see easy way to get the conf dir over to `Fork_id`, except maybe by setting another ref, which is still a can of worms. So I'll stick with current strategy, which loads the fork id along with other config items at daemon startup.

The fact that this is the case smells like we need to refactor how we handle our configuration directory. But, let's consider it out of scope for this :man_shrugging:.